src/tokens/aoTokens/ao.ts

```diff
@@ -153,31 +166,47 @@ export function useAoTokens(): [TokenInfoWithBalance[], boolean] {
         const balances = await Promise.all(
           tokens.map(async ({ id }) => {
             try {
-              const balance = await timeoutPromise(
-                (async () => {
-                  if (id === AO_NATIVE_TOKEN) {
-                    const res = await dryrun({
-                      Id: "0000000000000000000000000000000000000000001",
-                      Owner: activeAddress,
-                      process: AO_NATIVE_TOKEN_BALANCE_MIRROR,
-                      tags: [{ name: "Action", value: "Balance" }]
-                    });
-                    const balance = res.Messages[0].Data;
-                    if (balance) {
-                      return new Quantity(
-                        BigInt(balance),
-                        BigInt(12)
-                      ).toString();
+              const balance = Number(
```

Comment:
balance should not be converted to Number as precision is lost with it. Please, keep it as string.